processor/src/operations/horner_ops.rs

```diff
@@ -0,0 +1,502 @@
+    /// the same instruction interleaved with any operation that loads the next batch of 4
+    /// coefficients on the top of the operand stack, i.e., `mem_stream` or `adv_pipe`.
+    ///
+    /// The stack transition of the instruction can be visualized as follows:
+    ///
+    /// Input:
+    ///
+    /// +------+------+------+------+------+------+------+------+---+---+---+---+---+----------+------+------+
+    /// | c0_1 | c0_0 | c1_1 | c1_0 | c2_1 | c2_0 | c3_1 | c3_0 | - | - | - | - | - |alpha_addr| acc1 | acc0 |
+    /// +------+------+------+------+------+------+------+------+---+---+---+---+---+----------+------+------+
+    ///
+    ///
+    /// Output:
+    ///
+    /// +------+------+------+------+------+------+------+------+---+---+---+---+---+----------+-------+-------+
+    /// | c0_1 | c0_0 | c1_1 | c1_0 | c2_1 | c2_0 | c3_1 | c3_0 | - | - | - | - | - |alpha_addr| acc1' | acc0' |
+    /// +------+------+------+------+------+------+------+------+---+---+---+---+---+----------+-------+-------+
+    ///
+    ///
+    /// Here:
+    ///
+    /// 1. ci for i in 0..=4 stands for the the value of the i-th coefficient in the current batch
+    ///    of 4 extension field coefficients of the polynomial.
+    /// 2. (acc0, acc1) stands for an extension field element accumulating the values of the Horner
+    ///    evaluation procedure. (acc0', acc1') is the updated value of this accumulator.
+    /// 3. alpha_addr is the memory address of the evaluation point i.e., alpha.
+    ///
+    /// The instruction also makes use of the helper registers to hold the value of
+    /// alpha = (alpha0, alpha1) during the course of its execution.
+    pub(super) fn op_horner_eval_ext(&mut self) -> Result<(), ExecutionError> {
+        // --- read the values of the coefficients, over the extension field, from the stack ------
+        let coef = self.get_coeff_as_quad_ext_elements();
+
+        // --- read the evaluation point alpha from memory ----------------------------------------
+        let alpha = self.get_evaluation_point()?;
+
+        // --- read the accumulator values from stack ---------------------------------------------
+        let acc_old = self.get_accumulator();
+
+        // --- compute the updated accumulator values ---------------------------------------------
+        let acc_new = coef.iter().rev().fold(acc_old, |acc, coef| *coef + alpha * acc);
+
+        // --- copy the stack ---------------------------------------------------------
+        self.stack.copy_state(0);
+
+        // --- update the accumulators ------------------------------------------------------------
+        self.stack.set(ACC_HIGH_INDEX, acc_new.to_base_elements()[1]);
+        self.stack.set(ACC_LOW_INDEX, acc_new.to_base_elements()[0]);
+
+        // --- set the helper registers -----------------------------------------------------------
+        self.populate_helper_registers(alpha);
```

Comment:
Similar nit re simplify the comments as above.

Reply:
Changed